services/server/test/integration/repository-handlers/files.spec.ts

```diff
@@ -30,24 +30,103 @@ describe("Verify repository endpoints", function () {
+        },
+        {
+          name: "metadata.json",
+          path: `contracts/full_match/1337/${chainFixture.defaultContractAddress}/metadata.json`,
+          content: res2.body.files[1].content,
+        },
+      ],
+    });
     const res3 = await agent.get(
       `/files/tree/${chainFixture.chainId}/${chainFixture.defaultContractAddress}`
     );
-    chai.expect(res3.body).has.a.lengthOf(2);
+    chai
+      .expect(res3.body)
+      .to.deep.equal([
+        `http://localhost:10000/contracts/full_match/1337/${chainFixture.defaultContractAddress}/sources/project:/contracts/Storage.sol`,
+        `http://localhost:10000contracts/full_match/1337/${chainFixture.defaultContractAddress}/metadata.json`,
+      ]);
     const res4 = await agent.get(`/files/contracts/${chainFixture.chainId}`);
-    chai.expect(res4.body.full).has.a.lengthOf(1);
+    chai.expect(res4.body).to.deep.equal({
+      full: [chainFixture.defaultContractAddress],
+      partial: [],
+    });
+  });
+
+  it("should fetch a .sol file of specific address", async function () {
+    const agent = chai.request.agent(serverFixture.server.app);
+    // Wait for the server to complete the previous contract verification
+    await waitSecs(1);
+    await agent
+      .post("/")
+      .field("address", chainFixture.defaultContractAddress)
+      .field("chain", chainFixture.chainId)
+      .attach("files", chainFixture.defaultContractMetadata, "metadata.json")
+      .attach("files", chainFixture.defaultContractSource, "Storage.sol");
+    const res = await chai
+      .request(serverFixture.server.app)
+      .get(
+        `/repository/contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/sources/project:/contracts/Storage.sol`
+      );
+    chai
+      .expect(res.text)
+      .to.equal(chainFixture.defaultContractSource.toString());
+  });
+
+  it("should fetch metadata.json of specific address", async function () {
+    const agent = chai.request.agent(serverFixture.server.app);
+    // Wait for the server to complete the previous contract verification
+    await waitSecs(1);
+    await agent
+      .post("/")
+      .field("address", chainFixture.defaultContractAddress)
+      .field("chain", chainFixture.chainId)
+      .attach("files", chainFixture.defaultContractMetadata, "metadata.json")
+      .attach("files", chainFixture.defaultContractSource, "Storage.sol");
+    const res = await chai
+      .request(serverFixture.server.app)
+      .get(
+        `/repository/contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/metadata.json`
+      );
+    chai
+      .expect(res.text)
+      .to.equal(chainFixture.defaultContractMetadata.toString());
   });
 
-  it("should fetch immutable-references.json of specific address", async function () {
+  it("should fetch immutable-references.json of specific address, and it should be available in /files/tree", async function () {
```

Comment:
I think we should check if the contructor-args.txt etc. are inside the `files/any` with the correct content instead of `files/tree`. That was the main problem